app/controllers/apps_controller.rb

```diff
@@ -146,6 +149,19 @@ def parse_notice_at_notices_or_set_default
     end
   end
 
+  def process_fingerprinter_choice
+    if params['other'] && params['other']['use_site_fingerprinter']
+      fingerprinter = params['other']['use_site_fingerprinter']
+      if fingerprinter == SiteConfig::CONFIG_SOURCE_SITE
+        app.update_attributes(
+            notice_fingerprinter: SiteConfig.document.notice_fingerprinter_attributes)
```

Comment:
You are right, I've switched `write_attributes`. As indicated in http://www.davidverhasselt.com/set-attributes-in-activerecord/ , `write_attributes` doesn't trigger a database access. I've tested and works as claimed. Accessing and modifying app.notice_fingerprinter directly doesn't work because MongoId doesn't get aware of the change and doesn't save it when executing `app.save`.